### PR TITLE
dbeaver/pro#1201 Allow disabling values completion

### PR DESCRIPTION
**Pull request by ShadelessFox**
image

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java [hunk 1]
```diff
@@ -150,6 +152,7 @@ protected Control createPreferenceContent(@NotNull Composite parent) {
             csInsertSpace = UIUtils.createCheckbox(assistGroup, SQLEditorMessages.pref_page_sql_completion_label_insert_space, null, false, 2);
             csSortAlphabetically = UIUtils.createCheckbox(assistGroup, SQLEditorMessages.pref_page_sql_completion_label_sort_alphabetically, null, false, 2);
             csShowServerHelpTopics = UIUtils.createCheckbox(assistGroup, SQLEditorMessages.pref_page_sql_completion_label_show_server_help_topics, SQLEditorMessages.pref_page_sql_completion_label_show_server_help_topics_tip, false, 2);
+            csShowValues = UIUtils.createCheckbox(assistGroup, SQLEditorMessages.pref_page_sql_completion_label_show_values, SQLEditorMessages.pref_page_sql_completion_label_show_values_tip, false, 2);
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: It does not need to be false by default, IMHO.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: I agree with LonwoLonwo.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: It's assigned later to the value from the preference store:

```java
csShowValues.setSelection(store.getBoolean(SQLPreferenceConstants.SHOW_VALUES));
```

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: Ok, but wy it has an unchecked status here? What is the reason for this inconsistency?

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: Who cares about its initial value? I just copied the line above.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: I care. This is one of the places where I can see the default value. I need an extra click to find SQLPreferenceConstants and its default value.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCompletion.java:155]: Well, the initial state of a GUI control is not supposed to represent the initial value of the preference it modifies and you must not rely on it, so in the end, you will have to find where the preference's default value is initialized.

**Comment by uslss** [comment]
verified

**Pull request merged**